runtime/io.c

```diff
@@ -825,8 +838,7 @@ CAMLprim value caml_ml_flush(value vchannel)
   struct channel * channel = Channel(vchannel);
 
   caml_channel_lock(channel);
-  if (channel->fd != -1)
-    caml_flush(channel);
+  while (! flush_partial(channel, false)) /*nothing*/;
```

Comment:
```suggestion
  while (! flush_partial(channel, false)) continue;
```
is the preferred style and appeases most of the static analysis tools which are eager to complain about empty statements.

Reply:
Yes, but note that this is an exact analog of the code in `caml_flush` just above. So I would rather keep it like this or change both (with a preference for the former).